OBPIH-4704 Menu restyling (React)

kchelstowski opened the pull request.

drodzewicz commented on src/js/utils/MenuUtils.jsx:
```diff
@@ -0,0 +1,70 @@
+
+import _ from 'lodash';
+
+export function getMenuItemsComponent(section, key, active = false) {
```
To me, all of these utill functions in `MenuUtils.jsx` like:

- _getMenuItemsComponent_ 
- _getSubsectionComponent_ 
- _getSectionComponent_ 
- _getConfigurationSubsections_

They should not be utills but a simple function components with props:
- `<MenuItem />`
- `<MenuSection />`
- `<MenuSubsection />`
- `<MenuConfigurationSubsection />`

and they should be in components directory

kchelstowski replied:
nice idea, I just wanted to use already existing code wherever it was possible and it was one of them, as Menu was done like that:
```
 { _.map(menuConfig, (section, key) => {
          if (section.href) {
            return getSectionComponent(section, key);
          }
          if (section.subsections) {
            return getSubsectionComponent(section, key);
          }
          if (section.menuItems) {
            return getMenuItemsComponent(section, key);
          }
          return null;
        })
```
and those functions were at the top of the Menu component, so I wanted at least to move them to a separate file, but I'll probably proceed with your suggestion

kchelstowski commented on src/js/components/Layout/Menu.jsx:
```diff
@@ -1,77 +1,78 @@
+
+
+const Menu = ({ menuConfig, location }) => {
+  const sectionsToExclude = ['Analytics', 'Configuration'];
```
At first side I didn't want to do it on the backend side, as I wanted to avoid touching menu on gsp pages yet
Is `def getMenuConfig` in `ApiController` common for react and gsp? If it isn't, I could move it to backend.

kchelstowski replied:
I probably answered myself - I think that for gsp it uses `def megamenu` from `DashboardController`. If so, I'll try to move those `sectionsToExclude` to backend

drodzewicz replied:
I believe that `config.openboxes.megamenu` is a shared config between React and gsp.
There is also [`enabled` property on megamenu items](https://github.com/openboxes/openboxes/blob/develop/grails-app/conf/Config.groovy#L1403)  where you can probably set it to false and it will filter out this item from the menu

kchelstowski replied:
yeah, you're right, but if we don't want to touch gsp yet, we shouldn't make `enabled = false`, so I was just thinking about filtering it inside `def getMenuConfig`, so the megamenu would not be touched yet. 
Anyway we are going to remove it on gsp soon too, so maybe we could just proceed with this `enabled = false` even if those sections hide on gsp just now? 
@awalkowiak thoughts on that?

kchelstowski commented:
I found one bug connected to icons inside configuration dropdown - they didn't show at first render (they showed when I forced the project to bundle, for example using watcher and changing some random part of code). Refreshing the page both ctrl+r and ctrl+shift+r didn't change anything.
There was some problem with icon components, because when not working, printing out this:
```
const profileIcons = [
    <RiUser3Line />,
    <RiRefreshLine />,
    <RiLoginBoxLine />,
    <RiMapPinLine />,
  ];
``` 
showed this (take a look at the **type**):
![img](https://user-images.githubusercontent.com/93163821/188840556-221298df-a0ed-4e85-8f51-ef633c84b014.png)



and when working, it should look like this:

![img2](https://user-images.githubusercontent.com/93163821/188840509-84117c62-80c8-4c11-add7-62e60f685692.png)


I found "a fix" for it, but the `findIcon` function  now looks a bit uglier, but I haven't found any solution and we need to decide if we proceed with this solution or we should spend more time finding out why it didn't work in previous version (I do not know if it's a good idea to working it out now).

As for other fixes:
@drodzewicz I made `Analytics` section `enabled = false` in the `Config.groovy`, but I couldn't do the same with configuration, because I need to use it in the icons on the right side (I don't need it only in the main menu), so I decided to filter this one section on the frontend side.
I also changed those utils rendering sections/subsections to components as you proposed, removed `Navbar` component and changed colors to global variables.

awalkowiak reviewed (changes requested):
1. Can you just stop disabling eslint? You can just simply refactor the code in most cases.
2. Is impersonating users feature still work properly? How does logout behave for the impersonated user?
